pkg/engine/engine.go

```diff
@@ -158,6 +161,23 @@ func (e *Engine) alterFuncMap(t *template.Template) template.FuncMap {
 		return val, nil
 	}
 
+	// Add the 'tpl' function here
+	funcMap["tpl"] = func(tpl string, vals chartutil.Values) string {
+		r := renderable{
+			tpl:  tpl,
+			vals: vals,
+		}
+
+		templates := map[string]renderable{}
+		templates["template"] = r
+
+		result, err := e.render(templates)
+		if err != nil {
```

Comment:
Instead of returning the error as the result string I would favour making the function return `(string, error)` and failing the template execution altogether if the provided string does not render.

Otherwise errors are very hard to catch and will most certainly endup being deployed to k8s causing failures further down the road.

See the recently merged change for `include` https://github.com/kubernetes/helm/pull/2358 for some more details.